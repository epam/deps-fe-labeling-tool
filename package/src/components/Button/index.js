import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
