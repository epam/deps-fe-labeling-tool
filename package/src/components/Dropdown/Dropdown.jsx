
import React from 'react'
import { StyledDropdown } from './Dropdown.styles'

const Dropdown = (props) => <StyledDropdown {...props} />

export {
  Dropdown
}
