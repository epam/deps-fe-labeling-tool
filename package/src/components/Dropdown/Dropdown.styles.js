
import { Dropdown as AntdDropdown } from 'antd'
import 'antd/lib/dropdown/style/index.less'
import styled from 'styled-components'

const StyledDropdown = styled(AntdDropdown)`
  .ant-dropdown-menu-item {
    padding: 0;
  }
`

export {
  StyledDropdown
}
