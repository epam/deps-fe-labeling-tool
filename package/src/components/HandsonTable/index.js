
export * from './models'
export * from './HandsonTable'
