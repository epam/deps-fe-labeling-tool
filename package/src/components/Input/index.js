import { FilterInput } from './FilterInput'
import { Input } from './Input'
import { NumericInput } from './NumericInput'

Input.Numeric = NumericInput
Input.Filter = FilterInput

export {
  Input
}
