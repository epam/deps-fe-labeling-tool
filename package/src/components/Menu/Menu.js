
import { Menu } from 'antd'
import 'antd/lib/menu/style/index.less'

const MenuTrigger = {
  CLICK: 'click',
  CONTEXT_MENU: 'contextMenu',
  HOVER: 'hover'
}

export {
  Menu,
  MenuTrigger
}
