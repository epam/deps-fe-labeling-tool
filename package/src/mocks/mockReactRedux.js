const mockDispatch = jest.fn((action) => action)

const createMockMapStateToProps = (mapStateToProps) => {
  if (!mapStateToProps) {
    return mapStateToProps
  }

  return (state, ownProps) => ({
    props: mapStateToProps(state, ownProps)
  })
}

const getDispatchFunctionFormShorthandObject = (actionsObj) => (dispatch) => (
  Object.keys(actionsObj).reduce(
    (props, actionKey) => ({
      ...props,
      [actionKey]: jest.fn(
        (...args) => dispatch(actionsObj[actionKey].apply(undefined, args))
      )
    }),
    {}
  )
)

const createMockMapDispatchToProps = (mapDispatchToProps) => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps
  }

  const mappingFunction = typeof mapDispatchToProps === 'function'
    ? mapDispatchToProps
    : getDispatchFunctionFormShorthandObject(mapDispatchToProps)

  return (dispatch, ownProps) => {
    dispatch = dispatch || mockDispatch

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    }
  }
}

const createMockMergeProps = (mergeProps) => {
  if (!mergeProps) {
    return mergeProps
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: mockDispatch }

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    }
  }
}

const mockReactRedux = {
  connect: (mapStateToProps, mapDispatchToProps, mergeProps) => (WrappedComponent) => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    WrappedComponent
  }),
  batch: (fn) => Promise.resolve(fn),
  useDispatch: jest.fn(() => mockDispatch),
  useSelector: jest.fn((selector) => selector())
}

export {
  mockReactRedux
}
