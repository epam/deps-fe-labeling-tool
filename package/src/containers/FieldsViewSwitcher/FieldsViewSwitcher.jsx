
import React, { useCallback } from 'react'
import PropTypes from 'prop-types'
import { Button } from '@/components/Button'
import { Dropdown } from '@/components/Dropdown'
import { ViewOfDataIcon } from '@/components/Icons/ViewOfDataIcon'
import { Menu, MenuTrigger } from '@/components/Menu'
import { Placement } from '@/enums/Placement'

const RESOURCE_CHANGE_VIEW = 'CHANGE_VIEW'

class Group {
  constructor (code, name) {
    this.code = code
    this.name = name
  }
}

const groupShape = PropTypes.shape({
  code: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
})

const CHANGE_VIEW_TOOLTIP = {
  placement: Placement.TOP_RIGHT,
  title: RESOURCE_CHANGE_VIEW
}

const FieldsViewSwitcher = ({ groups, setFieldsGrouping }) => {
  const changeView = useCallback(
    (item) => {
      setFieldsGrouping(item.key)
    },
    [setFieldsGrouping]
  )

  const renderMenu = useCallback(
    () => (
      <Menu
        onClick={changeView}
        trigger={MenuTrigger.CLICK}
      >
        {
          groups.map((group) => (
            <Menu.Item
              key={group.code}
              id={group.code}
            >
              {group.name}
            </Menu.Item>
          ))
        }
      </Menu>
    ),
    [changeView, groups]
  )

  return (
    <Dropdown
      overlay={renderMenu()}
    >
      <Button.Icon
        icon={<ViewOfDataIcon />}
        tooltip={CHANGE_VIEW_TOOLTIP}
      />
    </Dropdown>
  )
}

FieldsViewSwitcher.propTypes = {
  groups: PropTypes.arrayOf(groupShape).isRequired,
  setFieldsGrouping: PropTypes.func.isRequired
}

export {
  Group,
  FieldsViewSwitcher
}
