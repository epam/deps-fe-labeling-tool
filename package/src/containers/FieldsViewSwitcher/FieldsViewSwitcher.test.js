
import { mockReactRedux } from '@/mocks/mockReactRedux'
import React from 'react'
import { shallow } from 'enzyme'
import { Dropdown } from '@/components/Dropdown'
import { Menu } from '@/components/Menu'
import { FieldsViewSwitcher, Group } from './FieldsViewSwitcher'

jest.mock('react-redux', () => mockReactRedux)

describe('FieldsViewSwitcher', () => {
  describe('component', () => {
    let defaultProps
    let wrapper

    beforeEach(() => {
      defaultProps = {
        groups: [new Group('test', 'Test')],
        setFieldsGrouping: jest.fn()
      }

      wrapper = shallow(<FieldsViewSwitcher {...defaultProps} />)
    })

    it('should render correct layout', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call props.setFieldsGrouping in case calling to Menu.props.onClick', () => {
      const DropdownProps = wrapper.find(Dropdown).props()
      const tempWrap = shallow(<div>{DropdownProps.overlay}</div>)
      const MenuProps = tempWrap.find(Menu).props()
      const mockItem = { key: 'test' }
      MenuProps.onClick(mockItem)
      expect(defaultProps.setFieldsGrouping).nthCalledWith(1, mockItem.key)
    })
  })
})
