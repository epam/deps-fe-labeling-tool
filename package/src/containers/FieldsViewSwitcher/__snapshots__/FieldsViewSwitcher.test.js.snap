// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldsViewSwitcher component should render correct layout 1`] = `
<Dropdown
  overlay={
    <Menu
      onClick={[Function]}
      trigger="click"
    >
      <MenuItem
        id="test"
      >
        Test
      </MenuItem>
    </Menu>
  }
>
  <IconButton
    icon={<ViewOfDataIcon />}
    tooltip={
      Object {
        "placement": "topRight",
        "title": "CHANGE_VIEW",
      }
    }
  />
</Dropdown>
`;
